### ISSUE TITLE
@JsonFormat.pattern on dates is now ignored if shape is not explicitely provided

### ISSUE THREAD

**Issue by fenrhil** [FasterXML/jackson-databind#1154]
Since #1111 was merged, `JsonFormat.pattern` is now only taken into account if the shape is set to `STRING`. When leaving the default shape (`ANY`), the pattern is ignored.

Thus this works:

```java
public class Person
{
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-mm-dd")
    public Date dateOfBirth;
}
```

But this doesn't (anymore):

```java
public class Person
{
    @JsonFormat(pattern = "yyyy-mm-dd") // Uses default shape, which is ANY
    public Date dateOfBirth;
}
```

From what I understand of [DataTimeSerializerBase](https://github.com/FasterXML/jackson-databind/blob/master/src/main/java/com/fasterxml/jackson/databind/ser/std/DateTimeSerializerBase.java)'s code, it seems that the default "shape" for a date-time is a String (see how `_asTimestamp` returns false by default). So it seems that providing a pattern should work with the shape `ANY`, which means "the default shape", which is `STRING`.

The issue stems from [this line](https://github.com/FasterXML/jackson-databind/pull/1111/files#diff-85c57322319123ea4bd790d52cf7aae8R62) . It should be something like this instead:

```java
if (format.getShape() == JsonFormat.Shape.STRING || format.getShape() == JsonFormat.Shape.ANY) {
```

PR coming as soon as I have the issue ID.

**Comment by cowtowncoder** [comment]
Thank you for reporting this. I think you are probably right with your assessment and this was just an oversight.

**Comment by cowtowncoder** [comment]
Fixed for 2.7.3.
